src/tiled/mainwindow.cpp

```diff
@@ -1509,6 +1493,98 @@ void MainWindow::retranslateUi()
     CommandManager::instance()->retranslateUi();
 }
 
+void MainWindow::exportMapAs(MapDocument *mapDocument)
+{
+    QString fileName = mapDocument->fileName();
+    QString selectedFilter =
+            mSettings.value(QLatin1String("lastUsedExportFilter")).toString();
+    auto exportDetails = chooseExportDetails<MapFormat>(fileName,
+                                                        mapDocument->lastExportFileName(),
+                                                        selectedFilter,
+                                                        this);
+    if (!exportDetails.isValid()) {
+        return;
+    }
+
+    // Check if writer will overwrite existing files here because some writers
+    // could save to multiple files at the same time. For example CSV saves
+    // each layer into a separate file.
+    QStringList outputFiles = exportDetails.mFormat->outputFiles(mapDocument->map(), exportDetails.mFileName);
+    if (outputFiles.size() > 0) {
+        // Check if any output file already exists
+        QString message =
+                tr("Some export files already exist:") + QLatin1String("\n\n");
+
+        bool overwriteHappens = false;
+
+        for (const QString &outputFile : outputFiles) {
+            if (QFile::exists(outputFile)) {
+                overwriteHappens = true;
+                message += outputFile + QLatin1Char('\n');
+            }
+        }
+        message += QLatin1Char('\n') + tr("Do you want to replace them?");
+
+        // If overwrite happens, warn the user and get confirmation before exporting
+        if (overwriteHappens) {
+            const QMessageBox::StandardButton reply = QMessageBox::warning(
+                                                          this,
+                                                          tr("Overwrite Files"),
+                                                          message,
+                                                          QMessageBox::Yes | QMessageBox::No,
+                                                          QMessageBox::No);
+
+            if (reply != QMessageBox::Yes)
+                return;
+        }
+    }
+
+    Preferences *pref = Preferences::instance();
+
+    pref->setLastPath(Preferences::ExportedFile, QFileInfo(exportDetails.mFileName).path());
+    mSettings.setValue(QLatin1String("lastUsedExportFilter"), selectedFilter);
+
+    auto exportResult = exportDetails.mFormat->write(mapDocument->map(), exportDetails.mFileName);
+    if (!exportResult)
+    {
```

Comment:
Coding style: please put `{` on the same line as the `if`

Reply:
Done